src/random.cpp

```diff
@@ -36,9 +36,13 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/aux_/disable_warnings_push.hpp"
 
+#ifdef BOOST_NO_CXX11_HDR_RANDOM
```

Comment:
yeah, I guess other parts of the code checks the __cplusplus define, that's probably the best to use. The reason I would be a bit worried about the boost define is that ``boost/config.hpp`` may not always have been included at this point